Sources/xcodeproj/Objects/Project/PBXObjectParser.swift

```diff
@@ -63,7 +63,7 @@ final class PBXObjectParser {
         case PBXBuildRule.isa:
             return try decoder.decode(PBXBuildRule.self, from: data)
         default:
-            throw PBXObjectError.unknownElement(isa)
+            preconditionFailure("unsupported object type")
```

Comment:
Considering how often new objects are introduced to Xcode projects, I think it's safe to go with a `preconditionFailure` avoiding the need of having to catch errors.